programs/admin.py

```diff
@@ -24,6 +27,69 @@
 )
 
 
+class ProgramNavigatorInline(TabularInline):
+    """
+    Sortable inline for managing Navigator ordering within a Program.
+    Uses Unfold's drag-and-drop sortable functionality.
+    """
+
+    model = ProgramNavigator
+    extra = 0
+    ordering_field = "order"
+    hide_ordering_field = False
+    fields = ["navigator", "order"]
+    autocomplete_fields = ["navigator"]
+
+    def get_queryset(self, request):
+        """Optimize queries with select_related"""
+        qs = super().get_queryset(request)
+        return qs.select_related("navigator", "program")
+
+    def get_formset(self, request, obj=None, **kwargs):
+        """
+        Auto-assign next sequential order to new navigators.
+        Ensures consistent sequential numbering (0, 1, 2, 3...) instead of gaps.
+
+        Edge cases handled:
+        - New program with no navigators: starts at 0
+        - Empty queryset: safely defaults to 0
+        - Multiple concurrent adds: each gets incremented initial value
+        - After drag-and-drop renumbering: continues from max
+        """
+        formset = super().get_formset(request, obj, **kwargs)
+
+        if obj:  # Only for existing programs (not new program creation)
+            try:
+                # Get the highest current order value from existing navigators
+                max_order_result = obj.program_navigators.aggregate(Max("order"))
+                max_order = max_order_result.get("order__max")
+
+                # Set initial value for new items
+                if max_order is not None:
+                    # Continue sequence from highest existing order
+                    formset.form.base_fields["order"].initial = max_order + 1
+                else:
+                    # First navigator for this program
+                    formset.form.base_fields["order"].initial = 0
+            except (AttributeError, KeyError):
+                # Fallback to 0 if anything goes wrong
+                formset.form.base_fields["order"].initial = 0
+
+        return formset
+
+    def formfield_for_foreignkey(self, db_field, request, **kwargs):
```

Comment:
If I'm understanding correctly, not sure if this is working for me. I see all white label navigators in the dropdown, but I'm using a super user so not sure if that affects it. From what I can tell, the previous functionality doesn't filter on white label either so maybe it's not a big deal. 

<img width="724" height="458" alt="Image" src="https://github.com/user-attachments/assets/e3e4710e-e95c-435b-867c-da3865820f5c" />